spring-integration-file/src/main/java/org/springframework/integration/file/tail/OSDelegatingFileTailingMessageProducer.java

```diff
@@ -110,7 +109,7 @@ private void runExec() {
 			logger.info("Starting tail process");
 		}
 		try {
-			Process process = Runtime.getRuntime().exec(this.command);
+			Process process = Runtime.getRuntime().exec(getCommand());
```

Comment:
How about to follow here with the same `reconfiguration` rule like in the `ApacheCommonsFileTailingMessageProducer`?
To change the `file` you should `stop` adapter, set a new value for the `file` and `start` adapter afterward.

Hence it would be better to move the `command` "compilation" to the `OSDelegatingFileTailingMessageProducer#doStart()` before `this.getTaskExecutor().execute`.

WDYT?

Seems for me, having such a change, there won't be need to fix that broken `FileTailInboundChannelAdapterParserTests` :smile: